src/binary_search_tree.js

```diff
@@ -0,0 +1,192 @@
+class Node {
+  constructor() {
+    this.parent = null
+    this.rightChild = null
+    this.leftChild = null
+    this.data
+  }
+}
+//////HELPER FUNCTIONS
+function hasChildren(node) {
+  return node.rightChild || node.leftChild
+}
+
+function recursiveInsert(startNode,newNode) {
+  if(newNode.data<=startNode.data) {
+    if(!startNode.leftChild) {
+      startNode.leftChild = newNode
+      newNode.parent = startNode
+    }
+    else {
+      return recursiveInsert(startNode.leftChild,newNode)
+    }
+  }
+  else{
+    if(!startNode.rightChild) {
+      startNode.rightChild = newNode
+      newNode.parent = startNode
+    }
+    else {
+      return recursiveInsert(startNode.rightChild,newNode)
+    }
+  }
+}
+
+function recursiveSearch(searchNode,value) {
+  if(value===searchNode.data) {
+    return searchNode
+  }
+  if(value < searchNode.data) {
+    if(searchNode.leftChild) {
+      return recursiveSearch(searchNode.leftChild, value)
+    }
+  }
+  else {
+    if(searchNode.rightChild) {
+      return recursiveSearch(searchNode.rightChild, value)
+    }
+  }
+  return null
+}
+
+function recursiveCount(node,count=0) {
+
+  if(node.leftChild) count = recursiveCount(node.leftChild,count+1)
```

Comment:
this is very non intuitive. The idiomatic way to write recursive is to write the exit condition of the recursion right at the beginning, so that way you dont have to set the value of `count`
In this function, that would look like -
`
function recursiveCount(node, count=0) {
  if(!node) return count
  else return (1 + recursiveCount(node.leftSubTree) + recursiveCount(node.rightSubtree)
`